.github/workflows/publish.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+    paths: [package.json]
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created:  ${{ steps.release.outputs.release_created }}
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          release-type: node
+          package-name: corepack
+
+  npm-publish:
+    needs: release-please
+    if: ${{ needs.release-please.outputs.release_created }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+
+      - name: Get the Yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(corepack yarn config get cacheFolder)"
+
+      - uses: actions/cache@v3
+        with:
+          path: ${{steps.yarn-cache-dir-path.outputs.dir}}
+          key: ${{runner.os}}-yarn-${{hashFiles('**/yarn.lock')}}
+          restore-keys: |
+            ${{runner.os}}-yarn-
+
+      - run:
+          corepack yarn install --immutable
+          corepack yarn npm publish
+        env:
+          YARN_NPM_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```

Comment:
Side note: it'd be nice to have this env variable documented in https://yarnpkg.com/cli/npm/publish, I didn't know about it and was using `corepack yarn config set npmAuthToken ${{ toJSON(secrets.NPM_TOKEN) }}` on my own repos.

Reply:
Almost all Yarn settings can be set via environment variables (we document it at the top of the [settings page](https://yarnpkg.com/configuration/yarnrc), but perhaps an explicit callout in the publish command would be worthwhile indeed 🤔)

Reply:
Ah yeah, I completely missed the paragraph about env variables. Maybe there could be a third tab named "Environment variable" which would say something like "Any settings can be set from env variable using this pattern, checkout `.yarnrc` tab for the full list of supported settings."